### PR TITLE
Feature/java 8 version

### PR DESCRIPTION
**Pull request by smic1909**
**Description**
This is the initial PR to upgrade from Java 7 to Java 8. First minor changes are already implemented. Further PRs can follow once this PR has been merged.

**Tested scenarios**
Tests are all green

**Fixed issue**:  #329

### CODE REVIEW

**Comment by martinsrenato** [comment]
Hi @smic1909,

Many thanks for the effort creating this pull request! 
This is a big chance that will be followed by a major release, so our team is reviewing it carefully and running some tests to see if everything still works as expected. We'll soon let you know if there's any comments regarding it.

Regards,
Renato
Adyen

**Comment by martinsrenato** [comment]
Hi @smic1909 ,

There's some conflicts on you pull request, can you please fix them?
They're really simple and related to a previous PR you created, so there should be no problem.

Regards,
Renato
Adyen

**Comment by smic1909** [comment]
@martinsrenato FYI, in case you missed my merge. Should be ok now, can you please double check.

File: src/main/java/com/adyen/model/checkout/DefaultPaymentMethodDetails.java [hunk 1]
```diff
@@ -237,22 +237,6 @@ public DefaultPaymentMethodDetails encryptedSecurityCode(String encryptedSecurit
         return this;
     }
 
-    /**
-     * @deprecated This field is deprecated. Use {@link DefaultPaymentMethodDetails#storedPaymentMethodId } instead.
-     */
-    @Deprecated
-    public String getRecurringDetailReference() {
```
[the remark this answers is not in the document]
Comment (smic1909) [reply src/main/java/com/adyen/model/checkout/DefaultPaymentMethodDetails.java]: Hey @andrei-dediu,

as requested, I reverted the change I pushed before. Please have a look.

**Comment by martinsrenato** [comment]
Hi @smic1909 !

Sorry for taking some time to answer you.
Your changes look fine and we already approved, but before merging we would like to ask if it's possible for you to commit one last change.
As you can see, the Travis build for this PR is failing. Currently it is building for both java 7 and 8, and after your change it is logically breaking for the java 7 build.
Could you please change the [`travis.yml`](https://github.com/Adyen/adyen-java-api-library/blob/develop/.travis.yml) file to remove java 7 as one of the JDKs to build? We would also like to include java 11 on it, so basically keep the `jdk` session of the file as
```
jdk:
  - oraclejdk8
  - oraclejdk11
```
Since the changes were made on your cloned repo, it is easier if you commit it. After that Travis will build again, hopefully without any issues.

Thanks again for all the contributions.

Regards,
Renato
Adyen

**Comment by martinsrenato** [comment]
Hi @smic1909 !

Sorry for taking some time to answer you.
Your changes look fine and we already approved, but before merging we would like to ask if it's possible for you to commit one last change.
As you can see, the Travis build for this PR is failing. Currently it is building for both java 7 and 8, and after your change it is logically breaking for the java 7 build.
Could you please change the [`travis.yml`](https://github.com/Adyen/adyen-java-api-library/blob/develop/.travis.yml) file to remove java 7 as one of the JDKs to build? So basically keep the `jdk` session of the file as
```
jdk:
  - oraclejdk8
```
Since the changes were made on your cloned repo, it is easier if you commit it. After that Travis will build again, hopefully without any issues.

Thanks again for all the contributions.

Regards,
Renato
Adyen

**Comment by smic1909** [comment]
> Hi @smic1909 !
> 
> Sorry for taking some time to answer you.
> Your changes look fine and we already approved, but before merging we would like to ask if it's possible for you to commit one last change.
> As you can see, the Travis build for this PR is failing. Currently it is building for both java 7 and 8, and after your change it is logically breaking for the java 7 build.
> Could you please change the [`travis.yml`](https://github.com/Adyen/adyen-java-api-library/blob/develop/.travis.yml) file to remove java 7 as one of the JDKs to build? So basically keep the `jdk` session of the file as
> 
> ```
> jdk:
>   - oraclejdk8
> ```
> 
> Since the changes were made on your cloned repo, it is easier if you commit it. After that Travis will build again, hopefully without any issues.
> 
> Thanks again for all the contributions.
> 
> Regards,
> Renato
> Adyen

Hi @martinsrenato , please check my latest commit.

**Comment by martinsrenato** [comment]
Hi @smic1909,

I am merging your PR now, since all the issues have been fixed and approved by reviewers.
This will be included on our next major release, which should happen on the next coming days.

Thanks again for the contribution!

Regards,
Renato
Adyen

**Pull request merged**